src/crafting.cpp

```diff
@@ -1625,12 +1670,12 @@ void player::complete_craft()
     // If we're here, the craft was a success!
     // Use up the components and tools
     std::list<item> used;
-    for (const auto &it : making->requirements.components) {
-        std::list<item> tmp = consume_items(it, batch_size);
+    for (int i = 0; i < making->requirements.components.size(); i++) {
+        std::list<item> tmp = consume_items(comp_selections[i], batch_size); // again, we consume the items we selected earlier
         used.splice(used.end(), tmp);
     }
-    for (const auto &it : making->requirements.tools) {
-        consume_tools(it, batch_size);
+	for (int i = 0; i < making->requirements.tools.size(); i++) {
```

Comment:
Any reason you're changing all those `for` loops into indexed ones?
Couldn't you iterate over `tool_selections` here? That would be more readable than having max index depend on `making->requirements.tools.size()`, but the actual indexed variable be from `tool_selections`.